front-end/src/app/new-company/new-company.component.html

```diff
@@ -0,0 +1,87 @@
+<div class="form-wrapper">
+  <h2>Register your company</h2>
+  <form [formGroup]="companyFormGroup" (ngSubmit)="createCompany()">
+    <mat-form-field>
+      <input
+        type="text"
+        formControlName="name"
+        placeholder="Company name"
+        required
+        matInput />
+      @if (companyFormGroup.invalid) {
+        <mat-error>This field is required</mat-error>
+      }
+    </mat-form-field>
+    <mat-form-field>
+      <input
+        type="text"
+        formControlName="location"
+        placeholder="Company location"
+        required
+        matInput />
+      @if (companyFormGroup.invalid) {
+        <mat-error>This field is required</mat-error>
+      }
+    </mat-form-field>
+    <mat-form-field>
+      <input
+        type="number"
+        formControlName="founded"
+        placeholder="Foundation year"
+        required
+        matInput />
+      @if (companyFormGroup.invalid) {
+        <mat-error>This field is required</mat-error>
+      }
+    </mat-form-field>
+    <div class="company-size">
+      <mat-form-field>
+        <input
+          type="number"
+          formControlName="sizeMin"
+          placeholder="Minimum size"
+          required
+          matInput />
+        @if (companyFormGroup.invalid) {
+          <mat-error>This field is required</mat-error>
+        }
+      </mat-form-field>
+      -
+      <mat-form-field>
+        <input
+          type="number"
+          formControlName="sizeMax"
+          placeholder="Maximum size"
+          required
+          matInput />
+        @if (companyFormGroup.invalid) {
+          <mat-error>This field is required</mat-error>
+        }
+      </mat-form-field>
+    </div>
+    <div class="file-picker-wrapper">
+      <label for="file">Company Logo:</label>
+      <input
+        class="file-picker"
+        id="file"
+        type="file"
+        formControlName="companyLogo"
+        required
+        (change)="onFilePicked($event)"
+        accept="image/*" />
+      <div
+        *ngIf="
+          companyFormGroup.invalid &&
+          (companyFormGroup.dirty || companyFormGroup.touched)
+        ">
+        <div *ngIf="companyFormGroup.errors?.['required']">
+          You must upload a logo
+        </div>
+      </div>
```

Comment:
```suggestion
      <div
    *ngIf="
      companyFormGroup.invalid &&
      (companyFormGroup.dirty || companyFormGroup.touched) &&
      companyFormGroup.get('companyLogo')?.errors?.required
    ">
    You must upload a logo
</div>
```
or something like this can be done I think.